tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -1620,6 +1620,44 @@ TEST_P(RDFVary, ManyVariationsManyColumns)
    }
 }
 
+// Regression test for https://github.com/root-project/root/issues/17486
+TEST_P(RDFVary, CompatibleColumnTypes)
+{
+   struct DataRAII {
+      const char *fFileName{"rdf_vary_compatible_column_types.root"};
+      const char *fTreeName{"tree"};
+      DataRAII()
+      {
+         TFile f{fFileName, "recreate"};
+         TTree t{fTreeName, fTreeName};
+         Float_t v[3] = {1.f, 2.f, 3.f};
+         t.Branch("v", &v, "v[3]/F");
+         t.Branch("w", &v, "w[3]/F");
+         t.Fill();
+         f.Write();
+      }
+      ~DataRAII() { std::remove(fFileName); }
+   } dataset;
+
+   ROOT::RDataFrame df_start{dataset.fTreeName, dataset.fFileName};
+   ROOT::RDF::RNode df{df_start};
+
+   // The JITted transformation triggers a change in the column type name
+   // due to the desugaring of the Float_t typedef
+   df = df.Redefine("v", "v");
+
+   // Previously Vary would throw when doing a string comparison check of the
+   // types of the columns to be varied
+   EXPECT_NO_THROW({
```

Comment:
Do we also have example where the exception in intentionally thrown?

Reply:
Added one 👍